packages/shared/src/utils/generate-joi.ts

```diff
@@ -215,7 +233,9 @@ export function generateJoi(filter: FieldFilter, options?: JoiOptions): AnySchem
 		}
 
 		if (operator === '_between') {
-			if (compareValue.every((value: any) => Number.isSafeInteger(Number(value instanceof Date ? NaN : value)))) {
+			if (
+				(compareValue as any).every((value: any) => Number.isSafeInteger(Number(value instanceof Date ? NaN : value)))
```

Comment:
`isSafeInteger` doesn't work on floats, so we should update this to check against `Number.MAX_SAFE_INTEGER`:

<img width="234" alt="CleanShot 2022-06-08 at 15 13 41@2x" src="https://user-images.githubusercontent.com/9141017/172698037-a4c4d20b-f432-491d-958b-a42634499a3a.png">